src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,20 +734,43 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
-    if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
-    ) {
-      throw new Error('Out of Story range.');
-    }
     if (storyDelta === 0) {
       return;
     }
-
-    const currentStory = this.stories_[this.currentIdx_ + storyDelta];
-
-    this.show(currentStory.href);
-    this.signalNavigation_();
+    if (this.element_.getAttribute('enable-circular-wrapping') !== 'true') {
+      if (
```

Comment:
Since we are performing this check pretty often around the codebase, it makes sense to extract this to a method `this.isIndexOutOfBounds_(index)` which returns if the new index is out of bounds.